lbrynet/core/client/ConnectionManager.py

```diff
@@ -146,57 +150,58 @@ def _send_secondary_requests(self, peer, protocol):
     @defer.inlineCallbacks
     def manage(self, schedule_next_call=True):
         self._manage_deferred = defer.Deferred()
-        if len(self._peer_connections) < conf.settings['max_connections_per_stream']:
+        if len(self._peer_connections) < self.max_connections_per_stream:
             log.debug("%s have %d connections, looking for %d",
                         self._get_log_name(), len(self._peer_connections),
-                        conf.settings['max_connections_per_stream'])
-            ordered_request_creators = self._rank_request_creator_connections()
-            peers = yield self._get_new_peers(ordered_request_creators)
-            new_conns = conf.settings['max_connections_per_stream'] - len(self._peer_connections)
-            peers = self._pick_best_peers(peers, new_conns)
+                        self.max_connections_per_stream)
+            peers = yield self._get_new_peers()
             for peer in peers:
                 self._connect_to_peer(peer)
         self._manage_deferred.callback(None)
         self._manage_deferred = None
         if not self.stopped and schedule_next_call:
             self._next_manage_call = utils.call_later(self.MANAGE_CALL_INTERVAL_SEC, self.manage)
 
-    def _rank_request_creator_connections(self):
-        """Returns an ordered list of our request creators, ranked according
-        to which has the least number of connections open that it
-        likes
-        """
-        def count_peers(request_creator):
-            return len([
-                p for p in self._peer_connections.itervalues()
-                if request_creator in p.request_creators])
-
-        return sorted(self._primary_request_creators, key=count_peers)
+    def return_shuffled_peers_not_connected_to(self, peers, new_conns_needed):
+        if peers is None:
+            # can happen if there is some error in the lookup
+            return []
+        out = [peer for peer in peers if peer not in self._peer_connections]
+        random.shuffle(out)
+        out = out[0:new_conns_needed]
+        return out
 
     @defer.inlineCallbacks
-    def _get_new_peers(self, request_creators):
+    def _get_new_peers(self):
+        new_conns_needed = self.max_connections_per_stream - len(self._peer_connections)
+        if new_conns_needed < 1:
```

Comment:
@jackrobison what if there's a bug and `new_conns_needed` ends up being -1 somehow? :smiling_imp: